pdns/common_startup.cc

```diff
@@ -563,6 +571,25 @@ void mainthread()
    DNSPacket::s_doEDNSSubnetProcessing = ::arg().mustDo("edns-subnet-processing");
    PacketHandler::s_SVCAutohints = ::arg().mustDo("svc-autohints");
 
+   if (::arg()["edns-cookie-secret"].size() != 0) {
+     // User wants cookie processing
+#ifdef HAVE_CRYPTO_SHORTHASH // we can do siphash-based cookies
+     DNSPacket::s_doEDNSCookieProcessing = true;
+     try {
+       if (::arg()["edns-cookie-secret"].size() != EDNSCOOKIESECRETSIZE) {
+         throw std::range_error("wrong size (" + std::to_string(::arg()["edns-cookie-secret"].size()) + "), must be " + std::to_string(EDNSCOOKIESECRETSIZE));
+       }
+       DNSPacket::s_EDNSCookieKey = makeBytesFromHex(::arg()["edns-cookie-secret"]);
+     } catch(const std::range_error &e) {
+       g_log<<Logger::Error<<"edns-cookie-secret invalid: "<<e.what()<<endl;
+       exit(1);
+     }
+#else
+     g_log<<Logger::Error<<"Support for EDNS Cookies is not available because of missing cryptographic functions"<<endl;
```

Comment:
```suggestion
     g_log<<Logger::Error<<"Support for EDNS Cookies is not available because of missing cryptographic functions (libsodium support should be enabled, with the crypto_shorthash() function available)"<<endl;
```